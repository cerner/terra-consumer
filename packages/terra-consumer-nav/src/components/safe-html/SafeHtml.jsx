import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  text: PropTypes.string.isRequired,
};

const SafeHtml = ({
  text,
  ...customProps
}) => (
  // eslint-disable-next-line react/no-danger
  <span {...customProps} dangerouslySetInnerHTML={{ __html: text }} />
);

SafeHtml.propTypes = propTypes;

export default SafeHtml;
