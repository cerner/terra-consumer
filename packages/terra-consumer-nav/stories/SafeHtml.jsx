import React from 'react';
import SafeHtml from '../src/components/safe-html/SafeHtml';

const textWithSpecChars = '&#8235;\u0627\u0644\u0639\u0631\u0628\u064a\u0651\u0629 &#8236;';
const textWithoutSpecChar = 'English (United States)';

const SafeHtmlTest = () => (
  <div>
    <h3>Text without special character: English (United States)</h3>
    <hr />
    <SafeHtml text={textWithoutSpecChar} />
    <hr />
    <h3>Text with special character: &amp;#8235;\u0627\u0644\u0639\u0631\u0628\u064a\u0651\u0629 &amp;#8236;</h3>
    <hr />
    <SafeHtml text={textWithSpecChars} />
    <hr />
  </div>
);

export default SafeHtmlTest;
