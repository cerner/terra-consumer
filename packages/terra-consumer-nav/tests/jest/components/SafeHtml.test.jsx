import React from 'react';
import SafeHtml from '../../../src/components/safe-html/SafeHtml';


describe('Render texts with special characters', () => {
  // Snapshot Tests
  it('should render a html element using dangerouslySetInnerHTML without any special character', () => {
    const wrapper = shallow(<SafeHtml text="English (United States)" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a html element using dangerouslySetInnerHTML with special characters ', () => {
    const wrapper = shallow(<SafeHtml text="&#8235;\u0627\u0644\u0639\u0631\u0628\u064a\u0651\u0629 &#8236;" />);
    expect(wrapper).toMatchSnapshot();
  });
});
