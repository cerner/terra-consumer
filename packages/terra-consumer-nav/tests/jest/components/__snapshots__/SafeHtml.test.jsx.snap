// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render texts with special characters should render a html element using dangerouslySetInnerHTML with special characters  1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "‫\\\\u0627\\\\u0644\\\\u0639\\\\u0631\\\\u0628\\\\u064a\\\\u0651\\\\u0629 ‬",
    }
  }
/>
`;

exports[`Render texts with special characters should render a html element using dangerouslySetInnerHTML without any special character 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "English (United States)",
    }
  }
/>
`;
